### PR TITLE
feat: add support for viewing favourite posts

### PR DESCRIPTION
**Pull request by david-hosier**
I have an idea to play around with a little app to view Mastodon favorites. I just came across your SDK today, and it looks pretty nice. I took a look, and it seemed simple enough to add support for viewing favorites to the SDK. Thanks for putting a great library; it looks like I can also learn a lot of new Swift ideas in general from looking at the code base.

I tried to regenerate the documentation per the contributing guidelines, but I get the following error:

```
TootSDK/Scripts on  main [!] took 1m1s
❯ ./updatedocs.sh
Running swift build
SPM build manifest does not exist at `/Users/david/dev/TootSDK/Scripts/.build/debug.yaml` or does not match expected format.
Error: Failed to generate documentation
/Users/david/.gem/gems/jazzy-0.14.3/lib/jazzy/executable.rb:39:in `execute_command': /Users/david/.gem/gems/jazzy-0.14.3/bin/sourcekitten ["doc", "--spm", "--module-name", "TootSDK", "--"] (RuntimeError)

Running swift build

SPM build manifest does not exist at `/Users/david/dev/TootSDK/Scripts/.build/debug.yaml` or does not match expected format.

Error: Failed to generate documentation
	from /Users/david/.gem/gems/jazzy-0.14.3/lib/jazzy/sourcekitten.rb:269:in `run_sourcekitten'
	from /Users/david/.gem/gems/jazzy-0.14.3/lib/jazzy/doc_builder.rb:82:in `block in build'
	from /Users/david/.gem/gems/jazzy-0.14.3/lib/jazzy/doc_builder.rb:80:in `chdir'
	from /Users/david/.gem/gems/jazzy-0.14.3/lib/jazzy/doc_builder.rb:80:in `build'
	from /Users/david/.gem/gems/jazzy-0.14.3/bin/jazzy:16:in `<top (required)>'
	from /Users/david/.gem/bin/jazzy:23:in `load'
	from /Users/david/.gem/bin/jazzy:23:in `<main>'

```

I do not have a .build/debug.yaml file under Scripts, and I've never used jazzy before, so I wasn't sure how to proceed.

### CODE REVIEW

**Comment by kkostov** [comment]
@david-hosier It seems the script expects to be run from the root of the repository, e.g. `sh ./Scripts/updatedocs.sh`

File: Sources/TootSDK/TootClient/TootClient+TimeLine.swift [hunk 1]
```diff
@@ -28,24 +23,33 @@ public extension TootClient {
...
-        let (data, response) = try await fetch(req: req)
-        let decoded = try decode([Post].self, from: data)
-        var pagination: Pagination?
-        
-        if let links = response.value(forHTTPHeaderField: "Link") {
-            pagination = Pagination(links: links)
+        return try await getPosts(req, pageInfo, limit)
+    }
+    
+    /// View favourite posts.
```
Comment (kkostov) [inline Sources/TootSDK/TootClient/TootClient+TimeLine.swift:45]:
```suggestion
    /// View posts that the user has favourited
```

File: Sources/TootSDK/TootClient/TootClient+TimeLine.swift [hunk 2]
```diff
@@ -9,12 +9,7 @@ import Foundation
 
 public extension TootClient {
     
-    func getHomeTimeline(_ pageInfo: PagedInfo? = nil, limit: Int? = nil) async throws -> PagedResult<[Post]> {
-        let req = HTTPRequestBuilder {
-            $0.url = getURL(["api", "v1", "timelines", "home"])
-            $0.method = .get
-            $0.query = getQueryParams(pageInfo, limit: limit)
-        }
+    private func getPosts(_ req: HTTPRequestBuilder, _ pageInfo: PagedInfo? = nil, _ limit: Int? = nil) async throws -> PagedResult<[Post]> {
```
Comment (kkostov) [inline Sources/TootSDK/TootClient/TootClient+TimeLine.swift:12]:
```suggestion
    internal func getPosts(_ req: HTTPRequestBuilder, _ pageInfo: PagedInfo? = nil, _ limit: Int? = nil) async throws -> PagedResult<[Post]> {
```

**Comment by kkostov** [comment]
Nicely done @david-hosier !

**Comment by davidgarywood** [comment]
Yup this is a good, clean PR - thank you @david-hosier !

Very happy to approve an merge this once docs are regenerated 👍

**Comment by david-hosier** [comment]
Thanks guys. I committed the suggested changes and ran the documentation update. Is it normal for most/all of the documentation files to become dirty after that process? I want to make sure before I add that to the PR.

**Comment by kkostov** [comment]
You’re right. I think it’s temporary that docs need to be generated manually, I’d like there to be an action to update them from main (eventually). Feel free to push the changes for now.

**Comment by david-hosier** [comment]
Makes sense, thanks. Doc updates have been pushed.

**Pull request merged**